tests/test_pulse.py

```diff
@@ -0,0 +1,31 @@
+import numpy as np
+import pytest
+from inline_snapshot import snapshot
+
+from promdens.promdens import ENVELOPE_TYPES, LaserPulse
+
+def test_invalid_envelope_type(make_pulse):
+    with pytest.raises(ValueError):
+        make_pulse(envelope_type='invalid')
+
+
+@pytest.mark.parametrize("envelope", ENVELOPE_TYPES)
```

Comment:
Instead of writing separate test function for each envelope type, here pytest auto-generates them for us by iterating over ENVELOPE_TYPES and passing the value via the "envelope" function parameter. You can see the individual tests by running pytest in verbose mode

```console
pytest -v
```